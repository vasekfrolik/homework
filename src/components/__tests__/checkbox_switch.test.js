import React from "react";
import { shallow } from "enzyme";

import CheckboxSwitch from "./../checkbox_switch";

let wrapper = shallow(
  <CheckboxSwitch
    label="Foo"
    checked={true}
    onChange={() => console.log("Test")}
  />
);

describe("render CheckboxSwitch", () => {
  it("check label and colon", () => {
    expect(wrapper.find(".im-switch-label").text()).toEqual("Foo:");
  });

  it("check input checked state and textual value", () => {
    let checkbox = wrapper.find({ type: "checkbox" });
    expect(checkbox.props().checked).toEqual(true);
    expect(wrapper.find(".im-slider-value").text()).toEqual("Yes");
    wrapper.setProps({ checked: false });
    let checkboxModified = wrapper.find({ type: "checkbox" });
    expect(checkboxModified.props().checked).toEqual(false);
    expect(wrapper.find(".im-slider-value").text()).toEqual("No");
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
