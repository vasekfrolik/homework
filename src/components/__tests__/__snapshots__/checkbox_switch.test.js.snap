// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render CheckboxSwitch matches snapshot 1`] = `ShallowWrapper {}`;
